apps/emqx_bridge_pulsar/test/emqx_bridge_pulsar_impl_producer_SUITE.erl

```diff
@@ -1064,9 +1064,17 @@ t_cluster(Config) ->
     ?check_trace(
         begin
             Nodes = [N1, N2 | _] = start_cluster(Cluster),
+            %% wait until bridge app supervisor is up; by that point,
+            %% `emqx_config_handler:add_handler' has been called and the node should be
+            %% ready to create bridges.
+            NumNodes = length(Nodes),
+            {ok, _} = snabbkaffe:block_until(
+                ?match_n_events(NumNodes, #{?snk_kind := emqx_bridge_app_started}),
+                15_000
+            ),
```

Comment:
Hm, suspicious that we really need that. This is part of application start routine, which means we'll wait [here](https://github.com/emqx/emqx/blob/26c2ab0bef8450d7c94da70be3619ede637636d9/apps/emqx/test/emqx_common_test_helpers.erl#L901) for `emqx_bridge_app:start/2` to complete anyway. This is because `emqx_bridge` is one of [apps](https://github.com/emqx/emqx/blob/f97ed9bab95c0573ebec0a61c831be52141162e6/apps/emqx_bridge_pulsar/test/emqx_bridge_pulsar_impl_producer_SUITE.erl#L506) this testcase needs to be started in the cluster.

Reply:
This was just an attempt after observing a [timeout](https://github.com/emqx/emqx/pull/10890/commits/b6db43edf47c08ed187bde54ae1fcd8c1d70ae46) while waiting for the bridge start events from the second node (that joins node 1).

We could also just increase the timeout, but I was wondering if it could be some asynchronicity related to the node joining the cluster and restarting its apps, so I attempted to add this and see if it improves the flakiness.  :see_no_evil:

Reply:
> related to the node joining the cluster and restarting its apps

Ah, I always forget about that. We'll probably need to finally hide that "async and [not really idempotent](https://github.com/emqx/emqx/blob/2250ed7052430cfca091b45f20e13d49ab3894a9/apps/emqx_bridge/test/emqx_bridge_api_SUITE.erl#L147-L157) restarts after cluster join" behind `emqx_common_test_helpers:start_slave/2`. Its contract is broken anyway: evidently, apps in `start_apps` are not really started. 🥲